File: JavaProject/I_Receptionist.java
import java.util.Date;
import java.util.List;
public interface I_Receptionist {
	
	public void Make_reservation(String client_id, Date start_date, Date end_date, int room_number);
	public void Take_payment(String client_id, double value, boolean card);
	public void Register_client(String last_name, String first_name, String client_id, String email);
	public List<Room> Find_free_room(Date start_date, Date end_date, int capacity);
	public void Check_in_client(String reservation_id);
	public void Cancel_reservation(String reservation_id);
	public void Check_out_client(String reservation_id);
	public Client Find_client(String client_id);
	public void Call_menager(String place);
	public void report_need_for_cleaning(int room_number);
	public void report_need_for_repair(int room_number);
	public void report_need_for_security_intervention(int room_number);

}

File: CSharpProject/Hotel/Interfaces/IReceptionist.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HotelProject.Model
{
    interface IReceptionist
    {
        void MakeReservation(String clientID, DateTime startDate, DateTime endDate, int roomNumber);
        void TakePayment(String clientID, double value, bool card);
        void RegisterClient(String lastName, String firstName, String clientID, String email);
        Room[] FindFreeRoom(DateTime startDate, DateTime endDate, int capacity);
        void CheckInClient(String reservationID);
        void CancelReservation(String reservationID);
        void CheckOutClient(String reservationID);
        Client FindClient(String clientID);
        void CallManager(String place);
        void ReportNeedForCleaning(int roomNumber);
        void ReportNeedForRepair(int roomNumber);
        void ReportNeedForSecurityIntervention(int roomNumber);
    }
}
